File: app/src/main/java/tech/pauly/findapet/discover/AnimalListItemViewHolder.java
package tech.pauly.findapet.discover;

import android.support.v7.widget.RecyclerView;

import tech.pauly.findapet.databinding.ItemAnimalListBinding;

public class AnimalListItemViewHolder extends RecyclerView.ViewHolder {

    private ItemAnimalListBinding binding;

    public AnimalListItemViewHolder(ItemAnimalListBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(AnimalListItemViewModel viewModel) {
        binding.setViewModel(viewModel);
        binding.executePendingBindings();
    }
}

File: app/src/main/java/tech/pauly/findapet/discover/AnimalListItemViewHolder.kt
package tech.pauly.findapet.discover

import androidx.recyclerview.widget.RecyclerView

import tech.pauly.findapet.databinding.ItemAnimalListBinding

class AnimalListItemViewHolder(private val binding: ItemAnimalListBinding) : RecyclerView.ViewHolder(binding.root) {

    fun bind(viewModel: AnimalListItemViewModel) {
        binding.also {
            it.viewModel = viewModel
            it.executePendingBindings()
        }
    }
}
